Userland/Libraries/LibJS/Bytecode/Op.cpp

```diff
@@ -525,6 +526,50 @@ ThrowCompletionOr<void> Call::execute_impl(Bytecode::Interpreter& interpreter) c
     return {};
 }
 
+// 13.3.7.1 Runtime Semantics: Evaluation, https://tc39.es/ecma262/#sec-super-keyword-runtime-semantics-evaluation
+ThrowCompletionOr<void> SuperCall::execute_impl(Bytecode::Interpreter& interpreter) const
+{
+    auto& vm = interpreter.vm();
+    // 1. Let newTarget be GetNewTarget().
+    auto new_target = vm.get_new_target();
+
+    // 2. Assert: Type(newTarget) is Object.
+    VERIFY(new_target.is_object());
+
+    // 3. Let func be GetSuperConstructor().
+    auto* func = get_super_constructor(vm);
+
+    // 4. Let argList be ? ArgumentListEvaluation of Arguments.
+    MarkedVector<Value> arg_list { vm.heap() };
+    for (size_t i = 0; i < m_argument_count; ++i)
+        arg_list.append(interpreter.reg(m_arguments[i]));
+
+    // 5. If IsConstructor(func) is false, throw a TypeError exception.
+    if (!func || !Value(func).is_constructor())
```

Comment:
You shouldn't need to check `!func` as `Value(nullptr)` should give a value of null and that isn't a constructor.
But honestly this is probably faster so keeping it is fine as well.

Reply:
This is copied from the AST-interpreter

Reply:
This isn't the happy path anyway, so let's remove the `!func`

Reply:
Sure, will do